nitro_rpc/web/server.go

```diff
@@ -0,0 +1,55 @@
+package web
+
+import (
+	"context"
+
+	"github.com/erc7824/examples/nitro_rpc/proto"
+	"github.com/layer-3/clearsync/pkg/signer"
+)
+
+var _ proto.NitroRPC = (*Server)(nil)
+
+type Server struct {
+	mh          MethodHandler
+	stateSigner signer.Signer
+}
+
+type MethodHandler interface {
+	HandleCall(ctx context.Context, params []byte) ([]byte, error)
+}
+
+func NewServer(handler MethodHandler, stateSigner signer.Signer) *Server {
+	return &Server{
+		mh:          handler,
+		stateSigner: stateSigner,
+	}
+}
+
+func (s *Server) Call(ctx context.Context, req *proto.Request) (*proto.Response, error) {
+	result, err := s.mh.HandleCall(ctx, req.Req.GetParams())
+	if err != nil {
+		return nil, err
+	}
+
+	state := RPCState{
+		RequestID: req.Req.GetRequestId(),
+		Timestamp: req.Req.GetTimestamp(),
```

Comment:
Shouldn't we set a new timestamp here, so that an external party can calculate how much time a response took place to calculate?

Reply:
As I discussed with @mnaichuk timestamp will be later generated by server, so that is TBD